feat: SfInput add icon and messages + SfSelect add messages

AdamPawlinski opened the pull request:
# Related issue
Closes #1104 

# Scope of work
continue work from #1128
Add messages to SfInput (hint and success) and additional icon.
Add messages to SfSelect (hint and sueccess). 

# Screenshots of visual changes
![image](https://user-images.githubusercontent.com/32042425/84629904-5da71500-aeeb-11ea-9ab0-df76616ac512.png)

![image](https://user-images.githubusercontent.com/32042425/84629830-3fd9b000-aeeb-11ea-8cbb-7a98837fc553.png)


# Checklist

- [x] No commented blocks of code left
- [x] Run tests and docs
- [x] Self code-reviewed

AdamPawlinski commented:
Yes, you are right, I'll change that. Thx.

Michal-Dziedzinski commented on packages/shared/styles/components/atoms/SfInput.scss:
```diff
@@ -99,15 +110,16 @@
-        --icon-color: var(--c-primary);
         --input-password-icon-background: var(--c-primary);
-        --input-label-top: 0;
+        --input-label-top: 0;        
```
Interesting change :D

AdamPawlinski replied:
great don't you think :)

Michal-Dziedzinski commented on packages/shared/styles/components/atoms/SfInput.scss:
```diff
@@ -164,14 +177,29 @@
-          --input-bar-background: var(--c-danger);
-          --input-label-color: var(--c-danger);
-        }
+          --input-bar-background: var(--c-danger);   
```
next one :D

Michal-Dziedzinski commented on packages/vue/src/components/atoms/SfInput/SfInput.spec.js:
```diff
@@ -14,18 +14,18 @@ describe("SfInput.vue", () => {
-      errorMessage
-    );
-  });
+  // it("renders errorMessage slot with valid is false when passed", () => {
```
if it's commented then shouldn't we remove it?

AdamPawlinski replied:
I commented it out because test doesn't let me push changes. I think it should be prepared to test component after this PR.

AdamPawlinski replied:
I changed this test to try if it passes errorMessage, not slot.

Michal-Dziedzinski commented on packages/shared/styles/components/atoms/SfInput.scss:
```diff
@@ -236,7 +300,7 @@
           --input-bar-width: 0;
           --input-label-color: var(--c-secondary-variant);
-        }
+        }      
```
? :)

Michal-Dziedzinski commented on packages/vue/src/components/atoms/SfInput/SfInput.stories.js:
```diff
@@ -33,20 +37,35 @@ storiesOf("Atoms|Input", module)
       required: {
         default: boolean("required", true, "Props"),
-      },
+      },      
```
?? :D

Michal-Dziedzinski commented on packages/vue/src/components/atoms/SfInput/SfInput.stories.js:
```diff
@@ -100,20 +124,33 @@ storiesOf("Atoms|Input", module)
       required: {
         default: boolean("required", true, "Props"),
-      },
+      },      
```
??? :D

Michal-Dziedzinski commented on packages/vue/src/components/atoms/SfInput/SfInput.stories.js:
```diff
@@ -127,12 +164,94 @@ storiesOf("Atoms|Input", module)
         :valid="valid"
         :error-message="errorMessage"
-        :required="required"
+        :required="required"        
```
ok, last comment about it, maybe some prettier changes :P

AdamPawlinski replied:
yes I think so.

Michal-Dziedzinski commented on packages/vue/src/components/atoms/SfInput/SfInput.vue:
```diff
@@ -155,6 +199,50 @@ export default {
     isPassword() {
       return this.type === "password" && this.hasShowPassword;
     },
+    computedMessageSlotName() {
```
There are the same logic if-else statements. I think there are 2 solutions you can consider.
1. Create one computet property that returns an object like `if (this.errorMessage && !this.valid) {
        return {slotName: "show-error-message", message: this.errorMessage, messageClass: "sf-input__message--error"}}`
2. Create a function to which you pass parameters `superFunction(error, success, hint) {
 if (this.errorMessage && !this.valid) {
        return error;
      } else if (this.successMessage && this.valid) {
        return success;
      } else if (this.hintMessage) {
        return this.required ? hint : "";
      } else {
        return "";
      }
}`
And then you can just `
computedMessageSlotName() {
return superFunction("show-error-message","show-success-message", "show-hint-message")
},
    computedMessage() {
return superFunction(this.errorMessage,this.successMessage, this.hintMessage)
    },
    computedMessageClass() {
return superFunction("sf-input__message--error","sf-input__message--success", "sf-input__message--success")
    },`

Wdyt?

AdamPawlinski replied:
of course you are right, maybe I'll try second solution.

AdamPawlinski commented:
Fixes made

filrak commented:
- star should be red when there is error message
- change `successMessage` to `infoMessage` as there is no "success" in inputs
- change color of `infoMessage` to green

AdamPawlinski commented:
In my opinion star is red already if there is error.
![image](https://user-images.githubusercontent.com/32042425/87422747-7b929300-c5d9-11ea-955a-9fd77b944682.png) 

successMessage is also green according to my local version 
![image](https://user-images.githubusercontent.com/32042425/87423102-0ecbc880-c5da-11ea-8f58-2802b6e2259e.png)

I changed name from successMessage to infoMessage and fixed all stories.

AdamPawlinski commented:
Conflicts resolved, messages slot added to SfComponentSelect and selected prop fixed in SfSelect.

AdamPawlinski commented:
I resolved conflicts and btw also the bug with label overwriting native placehloder and other one with icon size not changing after using prop.

justyna-13 commented on packages/vue/src/components/molecules/SfSelect/SfSelect.vue:
```diff
@@ -66,38 +70,86 @@ export default {
   },
-  data() {
-    return {
-      selectedValue: "",
```
Why is it deleted? :) there are errors in the console regarding this value

AdamPawlinski replied:
You are right, I restored selectedValue.

AdamPawlinski commented:
It's updated and adjusted to new designs, finally.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/vuestorefront/storefront-ui?pullRequest=1238) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/vuestorefront/storefront-ui?pullRequest=1238) before we can accept your contribution.<br/><hr/>**Adam Pawliński** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/vuestorefront/storefront-ui?pullRequest=1238) it.</sub>

AdamPawlinski commented:
I think we should close it, cause we already have masseges in SfInput And SfSelect.
